### ISSUE TITLE
August Release and Sovereign cloud - blocked by Power Platform for Admin V2 connector

### ISSUE THREAD

**Issue by lingutla123** [microsoft/coe-starter-kit#8835]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

Hi Jenefer,

We are in GCC environment. Can we install August release?

Thank you,

### What solution are you experiencing the issue with?

None

### What solution version are you using?

March  2024

### What method are you using to get inventory and telemetry?

None

**Comment by Jenefer-Monroe** [comment]
Hello Unfortunately not yet. The new Power Platform for Admin V2 connector is not yet supported in sovereign clouds. The team is actively working on this. 

Apologies we had hoped to wait to ship until it worked there but we had an escalation and had to ship. 

I'll pin this issue so that we have one to refer people.

**Comment by osbornecc** [comment]
Hi @Jenefer-Monroe 
We are GCC and last week we were able to use the connector.  We tested the action to elevate to System Administrator and it worked for us.

**Comment by osbornecc** [comment]
The only kicker was if the Solution created the connection for us, it was a broken connection.  I had to create the connection manually then let the Solution use that connection.  I created it with the OAuth option (not Service Principal)

**Comment by Jenefer-Monroe** [comment]
oh nice! So that did work for you? I'll reach out to check on status.

**Comment by osbornecc** [comment]
Yes! The Admin | Sync Template v4 (Driver) ran and elevated to System Administrator where needed.  We also had a case opened so we were watching for updates ;-)

**Comment by Jenefer-Monroe** [comment]
Wonderful news!

**Comment by wdanhof** [comment]
Assume this is still an issue for GCCH, the connection login redirects to .com from .us

**Comment by gatekeeper1337** [comment]
> Assume this is still an issue for GCCH, the connection login redirects to .com from .us

Same here, GCCH for the connector switches to .com when creating the connection.

**Comment by mhripko** [comment]
Since Microsoft has now said they will NOT fix the PP for Admin V2 connector's broken OAUTH support in GCC and GCCH, what is the recommendation going forward.  If it is to use the September 2024 and later releases, can someone explain how to UPGRADE a pre-September version to September 2024 or later without having to start all over.  I find that the Core solution in 2024 Sept still wants to engage the PP for Admin V2 connector and it we've never been able to use it successfully in GCCH, so we remain crippled.  Any thoughts, anyone

**Comment by wdanhof** [comment]
> Since Microsoft has now said they will NOT fix the PP for Admin V2 connector's broken OAUTH support in GCC and GCCH, what is the recommendation going forward. If it is to use the September 2024 and later releases, can someone explain how to UPGRADE a pre-September version to September 2024 or later without having to start all over. I find that the Core solution in 2024 Sept still wants to engage the PP for Admin V2 connector and it we've never been able to use it successfully in GCCH, so we remain crippled. Any thoughts, anyone

Where was that stated at?

**Comment by Grant-Archibald-MS** [comment]
@mhripko I will follow up gain with connector team next week to look to confirm is there is an update on this.

**Comment by gatekeeper1337** [comment]
@Grant-Archibald-MS - Has there been any movement on this? It's been keeping us from fully moving forward with our Power Platform adoption.

**Comment by Grant-Archibald-MS** [comment]
@gatekeeper1337 I have followed up with the internal team responsible for this connector.

I do not have any further updates to share at this stage.

**Comment by mhripko** [comment]
I'm not sure if this gets to the right people but when we found out the connector did not work for GCC High we were told to open a ticket with Microsoft to force them to track it and to provide updates after a couple of weeks and after Microsoft had changed the page and some of the content slightly but still claim that it's supported you she High we were told by somebody up in the Microsoft chain that they changed their mind and they had no intention of fixing the oauth support for that connector and the only reason we could get from them was a kind of half-hearted mention that using a service principal supposedly works with gcci although we could never get it to work at our company

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by mhripko** [comment]
Puzzling part of all of this is that if you actually go to the connector itself in the like in the DLP policies and things like that you'll see it has API endpoints that if properly implemented would allow you to set things like oath for GCC high but sadly they have been grayed out ever since the Inception of the connector which makes no sense

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by gatekeeper1337** [comment]
March release fixed the connector for us to get it installed. Fumbling through the install and failing workflows at the moment. Progress.

**Comment by mhripko** [comment]
Thank you for the update.  If there is anything I Can do to help, please let me know as this is really impacting us where I work.  I am working for a company that is a government contractor in the GCCHigh environment.  We have a lot of Power platform development going on and recently went enterprise wide with essentially the enterprise/site wide license (power app prem per user) for everyone in addition to many premium power automate, desktop power bi, and dynamics licenses just to set the stage.  Originally started with an older 2022/24 Coe Setup that worked will and reliably under the security changes made mid last year where system admins (system administrator role) and even the office 365 Power Platform Admin role cannot see into new environments being created unless it was already in their previously.  Of course, the Power Platform for Admins V2 connector is the culprit as it does not work /authenticate properly unless you are in commercial domain (it's broken in all GCC and GCCHigh clouds because the OAUTH code doesn't allow us to point it back at the gcchigh namespace.  IT's a shame because seems like September and all of the releases tried in some way to work around or tell us that the PP for Admin V2 connector would work, or maybe could work with a service principal - which doesn't seem to be true, an also additional code workaround solutions I've downloaded between sept and now tried to add code or so something that ultimately didn't work.

After logging a Microsoft support ticket against the connector (PP for admin V2) our Microsoft POC manager got back and said product support isn't going to address the PP4adminv2 for GCC and GCCH and there was no timeframe and closed the ticket.  about that time I wrote to you and others and logged the request.

We basically gave up trying to monitor the new environments being autoprovisioned and manually tried to periodically add them to the "service account" that our Coe toolkit runs under (our coe runs as an old style "service account" SRVC-COE-PROD and is assigned the M365 role Power Platform Administrator - a Priv account .  We patiently waited and hoped eventually enough pressure would come to bear and a workaround or the PP4Adminv2 would be fixed for gcchigh.  In February/March we attempted and updated the tools to the latest core and everything else.  The pp4Adminv2 connector went in and actually didn't block the install, and even appeared to work periodically but we could never really count on it to have access to new environments as the flow that the connector calls (driver something or other) would take 8 days to run every time it was launched each day and always got hung up and never completed a pass thru the environments.  reported and logged as a bug 9718 Driver Escalation Check by mhripko on 3/20/2025

after another reinstall just for a test, the errors stopped for a few days on the driver escalation check, but it was clear that the tool still didn't have an accurate count of the number of environments and we had to manually go to the environment membership page and press the "add me" bootstrap/force self into an environment for it to start seeing it.  It worked for about a week and then around April 3rd MS did some big over the weekend update to many parts and components in the platform and now every time the pp4adminv2 is invoked we get a driver escalation check on just about every environment including ones it always could access previously.  Now it's kind of random. some it can see into and work with, but many of them now cause the failure on the step/action pbattini shows "Grants requesting tenant admin user the system admin role".  Basically it now at the point where no one trusts the environment count because the escalation checks occur seemingly in some random pattern depending on the day and therefore no one trusts the Power BI dashboards with the counts of environments, and apps, and flows and other entities and when we try to investigate the failure messages in the sync flow failures, we find driver escalation checks show up and name an environment that it says it could not access yet we find that we may actually have access that we previously maybe have manually given it to.  Others that we know it should have access to fail and when we investigate and signed in as the service account for the toolkit we choose the membership panel in a failing environment and find that the panel is empty except for the Add Me/bootstrap button, and once manually pressed, that service account and all of the others appear as if it has access again and has forgotten how to get in/authenticate/bootstrap itself up.

In March as ***@***.*** i reported this against the Fed toolkit and asked why it takes 8 days/cycle to do some sort of exponential timeout.  Pbattini replied by asking/verifying that my timeout/setting was defaulting and exponential and after the retries of not getting into an environment it eventually hit 8 days and time-out / failed the flow run.

In try to reply to pbattini, i accidentally pressed the wrong button and and it did some reply and close my issue when i didn't want to close the issue.  I added a note that i wasn't done investigating why its broken still but pbattini never reploed except to move the bug from the April 2025 toolkit to possibly the may 2025 version without any more conversation.

Recently with the enterprise license signing, my boss has suggested that we gather all data to raise to the highest levels of mgmt. within our MS power platform hierarchy to see what could be done or get some timeframe or ideas to get this working again.  in GCCHigh, we have no new admin2 pages (still just the old ones) but have heard "available now in all environments" which clearly is not true so we are crippled with no admin center v2, and a non functional Coe setup that cannot even provide us with anything approaching an accurate count of our environments because of the ppadminv2 connector oauth problem.  We just want to be able to trust that it's getting into all of the environments and doing the inventory and counting accurately environments, apps, flows, and other things it finds.  Especially given that the new Admin2 pages/center still is not available in GCC or gcchigh even though ms keeps saying "soon" and writing up their docs as if it's available and out there.

Does anyone have an accurate env count with apps and flows accuratrely counted in those ENVs. running on GCCHigh/GCC or any sovereign cloud?  Any tests or troubleshooting ideas to check the reliability of the pp4adminv2 connector and it's ability to bootstrap itself into new env's system admin roles


________________________________

**Comment by mhripko** [comment]
Grant, do you know if they are going to fix the Power Platform for Admin V2 CONNECTOR SO THAT IT PROPERLY AUTHENTICATES WITH oauth IN gcc/gcchigh.  We are on the March version of the Coe tool kit and it's been months since we've had any reliable inventory scans of environments, apps, and flows because it seems like we keep getting errors with the connector or the infamous Driver escalation (I filed a bug under ***@***.*** on it last month).  Back in July/august of last year of course, it all stopped working when they changed the security model to not grant system admins access to the environments.  Now we've been unable to get reliable inventory cycles to run.

Of course being in GCCHigh, we also don't have the Admin2 pages for looking at some of environment as the new Admin center is still not available in GCC/GCCHigh.

We are in GCC high. ***@***.*** ir ***@***.***
________________________________

**Comment by github1339** [comment]
Hello, I just noticed the "Power Platform for Admins V2" connector is available to enable in our gcc high tenant along with the new admin center. Is it possible newer releases of CoE can be installed in sovereign clouds now?

We are on v4.42 for core components and v3.25 for governance. If it is possible to install now, should we plan to step up to the current version through previous releases because it has been over a year since we have been able to upgrade? I can start a new issue for this if appropriate.

Thanks all. Happy New Year.

**Comment by AmarSaiRam** [comment]
Dear User,

We noticed that this issue has been pending for quite some time, and we have not received any recent updates from your end. In the absence of further communication, we are proceeding to close this ticket, assuming that your concerns have been addressed or resolved.
If you continue to experience any difficulties related to this matter, please feel free to open a new support ticket. This will help us ensure prompt and effective assistance, rather than reopening the new Issue.
Thank you for your understanding and cooperation.

Best Regards,
CoE Starter Kit Team
